cmd/nginx/flags.go

```diff
@@ -303,16 +307,37 @@ https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-g
 		config.RootCAFile = *rootCAFile
 	}
 
+	var err error
 	if (nginx.MaxmindLicenseKey != "" || nginx.MaxmindMirror != "") && nginx.MaxmindEditionIDs != "" {
-		if err := nginx.ValidateGeoLite2DBEditions(); err != nil {
+		if err = nginx.ValidateGeoLite2DBEditions(); err != nil {
 			return false, nil, err
 		}
-		klog.InfoS("downloading maxmind GeoIP2 databases")
-		if err := nginx.DownloadGeoLite2DB(); err != nil {
+		timeout := time.Now().Add(time.Duration(nginx.MaxmindRetriesTimeout) * time.Second)
+		klog.InfoS("trying to download maxmind GeoIP2 databases...")
+		for {
+			err = nginx.DownloadGeoLite2DB()
+			if err == nil {
+				break
+			}
+			if e, ok := err.(*url.Error); ok {
+				if e, ok := e.Err.(*net.OpError); ok {
+					if e, ok := e.Err.(*os.SyscallError); ok {
+						if e.Err == syscall.ECONNREFUSED {
+							tn := time.Now()
+							if tn.Before(timeout) {
+								klog.InfoS("will try to download maxmind GeoIP2 DB in next " + timeout.Sub(tn).String() )
+								time.Sleep(time.Second)
```

Comment:
Backoff would be better than a fixed time. Here's an example:
https://github.com/kubernetes/ingress-nginx/blob/b1c8e3047ba31d7ea78f6e0915d187db75230ba5/cmd/nginx/main.go#L250-L279

Reply:
Fixed